src/pages/PassportRedemption/GiveawayPopover.tsx

```diff
@@ -0,0 +1,237 @@
+}
+
+const GiveawayPopover = (props: Props) => {
+  const [showInfo, setShowInfo] = useState(false);
+  const [giveawayData, setData] = useState<GiveawayData>({
+    weekly_giveaway_entries: '',
+    number_of_visits_left: undefined
+  })
+  useEffect(() => {
+    async function setGiveawayCount(contactId: string) {
+      try {
+        const qtySellers = await getAllParticipatingSellers()
+          .then((res) => res.data.length);
+        const { weekly_giveaway_entries, unique_seller_tickets } = await getGiveawayTicketsForContact(contactId)
+          .then((res) => res.data);
+        setData({weekly_giveaway_entries, number_of_visits_left: qtySellers - unique_seller_tickets})
+      } catch (err) {
+        console.error(err);
+      }
+    }
+    if (props.showInstagram && showInfo) {
+      setGiveawayCount(props.contactId)
+    }
+  }, [props.showInstagram, showInfo, props.contactId])
+  return (
+    <PopoverContainer>
+      {props.showInstagram && showInfo && <PopoverContent>
+        <ToggleClosedButton onClick={() => setShowInfo(false)}>
+          <InstagramEnabled/>
+        </ToggleClosedButton>
+        <Title>
+          INSTAGRAM FOR GIVEAWAY ADDED
+        </Title>
+        <Divider/>
+        <GiveawayTextContainer>
+          <InfoText>
+            <TextBold>{giveawayData.weekly_giveaway_entries}</TextBold>
+            <GeneralText>
+              Weekly Giveaway Entries
+            </GeneralText>
+          </InfoText>
+
+          {giveawayData.number_of_visits_left !== 0 ?
+          <InfoText>
+            <TextBold>{giveawayData.number_of_visits_left}</TextBold>
+            <GeneralText>
+              Vendor Tickets away from a Grand Prize Giveaway Entry
+            </GeneralText>
+          </InfoText> :
+          <InfoText icon={true}>
+            <IconWrap>
+              <GreenCheck/>
+            </IconWrap>
+            <GeneralText>Grand Prize Giveaway Entered</GeneralText>
+          </InfoText>
+          }
+        </GiveawayTextContainer>
+        <SubText>
+        Weekly Giveaway winners are announced on Instagram each Monday during the month of September. Grand Prize winners will be announced on October 1st.
+        </SubText>
+        <CloseButton type='button' onClick={() => setShowInfo(false)}>
+          CLOSE
+        </CloseButton>
+      </PopoverContent>}
+      {
+        props.showInstagram ?
+        <InstagramButton onClick={() => setShowInfo(!showInfo)}>
+          <InstagramEnabled/>
+        </InstagramButton> :
+        <InstagramButton disabled={true}>
+          <InstagramDisabled/>
+        </InstagramButton>
+      }
+    </PopoverContainer>
+  )
+}
+
+export default GiveawayPopover
+
+const PopoverContainer = styled.div`
+  position: absolute;
+  right: 0;
+`;
+
+const InstagramButton = styled.div`
+  position: absolute;
+  right: ${(props:IgButtonProps) => props.disabled ? '8.5px' : '4px'};
+  top: ${(props:IgButtonProps) => props.disabled ? '-3px': '1px'};
```

Comment:
What's this weirdness for? If you need anything like this mind dropping a comment? This negative top makes me nervous

Reply:
The weirdness is because it renders either InstagramEnabled or the InstagramEnabled icon and that ends up being positioned differently. Thinking about it now, it's probably easier just to make two separate buttons for both enabled/ disabled states.